### PR TITLE
Fix for compiling with NO_ASN_TIME + doc fix

### PR DESCRIPTION
**Pull request by lealem47**
# Description

- Guard use of `LowResTimer()` with `NO_ASN_TIME`
- Remove STM32 documentation that is no longer applicable

# Testing

./configure CFLAGS="-DNO_ASN_TIME"

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by lealem47** [comment]
Jenkins retest this please

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]
Please guard the `LowResTimer` call in `SetupSession` with `NO_ASN_TIME` instead. `SetupSession` is meant to setup the session even when the internal cache is not compiled in.

**Comment by lealem47** [comment]
> Please guard the `LowResTimer` call in `SetupSession` with `NO_ASN_TIME` instead. `SetupSession` is meant to setup the session even when the internal cache is not compiled in.

Done @julek-wolfssl

**Review by dgarske — APPROVED** [review]

File: src/ssl.c [hunk 1]
```diff
@@ -14501,7 +14501,9 @@ void SetupSession(WOLFSSL* ssl)
     }
 #endif
     session->timeout = ssl->timeout;
+#ifndef NO_ASN_TIME
```
Comment (dgarske) [inline src/ssl.c:14504]: Did you check what bornOn is used for? Should it be set to 0 or something else??

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (lealem47) [reply src/ssl.c:14504]: When `NO_ASN_TIME` was defined, it didn't look bornOn was used anywhere. References to bornOn are guarded by `NO_SESSION_CACHE`, which is defined when `NO_ASN_TIME` is defined. I should change the fix to be more explicit and use the appropriate macro. Will make the change shortly ....

Comment (lealem47) [reply src/ssl.c:14504]: Done

Comment (dgarske) [reply src/ssl.c:14504]: Don't you still need `NO_ASN_TIME` in the macro check also? `#if !defined(NO_SESSION_CACHE) && !defined(NO_ASN_TIME)`?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (lealem47) [reply src/ssl.c:14504]: Completely correct, I was being dumb. I reverted the last commit.` #if !defined(NO_ASN_TIME)` should be enough as `NO_ASN_TIME` defines `NO_SESSION_CACHE`

**Review by dgarske — APPROVED** [review]

**Review by julek-wolfssl — APPROVED** [review]
Thanks @lealem47

**Pull request merged**